src/main/java/Duke.java

```diff
@@ -1,10 +1,151 @@
+        String command = userInput.toLowerCase().trim();
+        return command;
+    }
+
+    private static String read(){                                           // To shift to read class
+            Scanner in = new Scanner(System.in);
+            String userInput = parse(in.nextLine());
+            return userInput;
+    }
+
+    private static void addToList (String userInput){
+        String[] splitString = userInput.trim().split(" ", 2);
+        if (splitString[1] == "" || splitString[1] == " "){
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        switch (splitString[0]) {
+            case "todo":
+                ToDo newTodo = new ToDo(splitString[1]);
+                list.add(newTodo);
+                break;
+            case "event":
+                String[] eventDescription = splitString[1].split("/at ", 2);
+                Events newEvent = new Events(eventDescription[0], eventDescription[1]);
+                list.add(newEvent);
+                break;
+            case "deadline":
+                String[] deadlineDescription = splitString[1].split("/by ", 2);
+                DeadLines newDeadline = new DeadLines(deadlineDescription[0], deadlineDescription[1]);
+                list.add(newDeadline);
+                break;
+            default:
+                return;
+        }
+        dukeReply("Got it. I've added this task:\n" + list.get(list.size()-1).getFullStatus() + "Now you have " + list.size() + " tasks in the list.\n");
+    }
+
+
+    private static void showList() throws DukeException {
+        String taskList = new String();
+        if (list.size() == 0){
+            throw new DukeException();
+        }
+        int counter = 1;
+        for (Task task : list){
+            taskList = taskList.concat(counter + "." + task.getFullStatus());
+            counter++;
+        }
+        dukeReply(taskList);
+    }
+
+    private static void markDone(String userInput) throws DukeExceptionInvalidTaskNumberDone, DukeException {
+        if (userInput.length() < 5 || userInput.charAt(4) != ' ' || userInput.charAt(5) == ' ') { // Handling errors: When user types done1 done2 or done   2
+            throw new DukeExceptionInvalidTaskNumberDone();
+        }
+        String taskNumber = userInput.substring(userInput.indexOf("done")+5);
+        try {
+            list.get(Integer.parseInt(taskNumber) - 1).isDone = true;
+        }
+        catch (IndexOutOfBoundsException e){
+            throw new DukeException();
+        }
+        dukeReply("Nice! I've marked this task as done: \n" + list.get(Integer.parseInt(taskNumber) - 1).getFullStatus());
+    }
+
+    private static void dukeReply(String reply){
+        System.out.println(line + reply + line);
+    }
+
+    private static String commandIdentifier(String userInput){
+        if (userInput.contains("done"))  //priority 1
```

Comment:
Can consider using a switch case instead as number of cases is more than 5, switch case will implement it as a lookup table or hash list which is much faster than if else.